core/server/middleware/check-ssl.js

```diff
@@ -44,7 +44,7 @@ checkSSL = function checkSSL(req, res, next) {
                 forceAdminSSL: config.get('forceAdminSSL'),
                 configUrlSSL: config.get('urlSSL'),
                 configUrl: config.get('url'),
-                reqUrl: req.url
+                reqUrl: req.originalUrl || req.url
```

Comment:
This is a good question. I think it is a bit weird how express works, and unfortunately we didn't really understand how it worked when we started out.

In uncapitalise I added an example, which is true for the old setup where only blogApp was mounted. With multiple mounted apps, more of the url lives in `baseUrl` instead of in `url`:

```
req.originalUrl = /blog/ghost/signin/?asdAD=asdAS
req.url = /ghost/signin/?asdAD=asdAS
req.baseUrl = /blog
req.path =  /ghost/signin/
```

I think, unfortunately, unless we always test properly we will always have issues in this area, just as we have constant regressions and bugs with subdirectories for the same reason.